test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -5036,6 +5036,11 @@ TEST_F(AtenXlaTensorTest, TestOneHot) {
     torch::Tensor xla_output = torch::one_hot(xla_input, num_classes);
     AllEqual(output, xla_output);
   });
+
+  // TODO: PT one_hot impl employs item() which could be eliminated.
```

Comment:
I am not sure I parse this.
If we call `one_hot()` on an XLA device, we still get one `item()` call?

Reply:
Oh so we never get called as `one_hot` but only `scatter_`. So when we get called, the `item()` check is already done. https://github.com/pytorch/pytorch/blob/d1623f4cc9334ade7376b3685bb91d3071e3b418/aten/src/ATen/native/Onehot.cpp#L21

Reply:
Oh OK, we need to fix that.